transform-plugins/src/main/java/co/cask/hydrator/plugin/Decoder.java

```diff
@@ -0,0 +1,223 @@
+
+      // If format is not right, then we throw an exception.
+      if (params.length < 2) {
+        throw new IllegalArgumentException("Configuration '" + mapping + "' is incorrectly formed. " +
+                                             "Format should be <fieldname>:<decoder-type>");
+      }
+
+      String field = params[0];
+      String type = params[1].toUpperCase();
+      DecoderType eType = DecoderType.valueOf(type);
+
+      if (decodeMap.containsKey(field)) {
+        throw new IllegalArgumentException("Field " + field + " already has decoder set. Check the mapping.");
+      } else {
+        decodeMap.put(field, eType);
+      }
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    parseConfiguration(config.decode);
+    // Check if schema specified is a valid schema or no. 
+    try {
+      Schema.parseJson(config.schema);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    parseConfiguration(config.decode);
+    try {
+      outSchema = Schema.parseJson(config.schema);
+      List<Field> outFields = outSchema.getFields();
+      for (Field field : outFields) {
+        outSchemaMap.put(field.getName(), field.getSchema().getType());
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) throws Exception {
+    StructuredRecord.Builder builder = StructuredRecord.builder(outSchema);
+
+    Schema inSchema = in.getSchema();
+    List<Field> inFields = inSchema.getFields();
+
+    // Iterate through input fields. Check if field name is present 
+    // in the fields that need to be decoded, if it's not then write
+    // to output as it is. 
+    for (Field field : inFields) {
+      String name = field.getName();
+
+      // Check if output schema also have the same field name. If it's not 
+      // then throw an exception. 
+      if (!outSchemaMap.containsKey(name)) {
+        continue;
+      }
+
+      Schema.Type outFieldType = outSchemaMap.get(name);
+
+      // Check if the input field name is configured to be decoded. If the field is not
+      // present or is defined as none, then pass through the field as is. 
+      if (!decodeMap.containsKey(name) || decodeMap.get(name) == DecoderType.NONE) {
+        builder.set(name, in.get(name));
+      } else {
+        // Now, the input field could be of type String or byte[], so transform everything
+        // to byte[] 
+        byte[] obj = new byte[0];
+        if (field.getSchema().getType() == Schema.Type.STRING) {
```

Comment:
in the beginning, should we verify that the schema type will either be STRING or BYTES?
or else, throw an exception in the `else` case?